app/validators/users/profile-details.validator.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+/**
+ * Validates data submitted for the `/users/me/profile-details` page
+ * @module ProfileDetailsValidator
+ */
+
+const Joi = require('joi')
+
+/**
+ * Validates data submitted for the `/users/me/profile-details` page
+ *
+ * There is no validation on this at all - there probably should be.
+ *
+ * @param {object} payload - The payload from the request to be validated
+ *
+ * @returns {object} the result from calling Joi's schema.validate(). If any errors are found the
+ * `error:` property will also exist detailing what the issue is.
+ */
+function go(payload) {
+  const schema = Joi.object().keys({
+    name: Joi.string().max(100).allow('').messages({
+      'string.max': 'Name must be 100 characters or less'
+    }),
+    jobTitle: Joi.string().max(100).allow('').messages({
+      'string.max': 'Job title must be 100 characters or less'
+    }),
+    email: Joi.string().email().allow('').messages({
+      'string.email': 'Enter a valid email'
+    }),
+    tel: Joi.string().max(100).allow('').messages({
+      'string.max': 'Telephone number must be 100 characters or less'
+    }),
+    address: Joi.string().max(100).allow('').messages({
+      'string.max': 'Address must be 100 characters or less'
+    })
+  })
+
+  return schema.validate(payload, { abortEarly: false })
+}
```

Comment:
I've added some max length validation as there wasn't any validation on these at all - not sure what the correct business rules should be

Reply:
As discussed earlier, double-check the data to ensure we're not imposing anything that would cause existing data to be considered invalid.

Having the mechanism to allow us to validate is the key thing (because we should be). It's a minor thing to tweak it once available.

Reply:
I've looked through the existing records and they're all either test emails, old defra emails or actual `@environment-agency.gov.uk` so I've gone with just that as a valid domain.

I've also upped the address length to 300 as there were some that were over 100 characters.

But overall this data is very sparsely populated so this is largely guesswork to now.